lib/src/map/map_controller.dart

```diff
@@ -20,7 +21,16 @@ abstract class MapController {
   /// instance.
   ///
   /// Factory constructor redirects to underlying implementation's constructor.
-  factory MapController() = MapControllerImpl._;
+  factory MapController() = MapControllerImpl;
+
+  static MapController? maybeOf(BuildContext context) => context
+      .dependOnInheritedWidgetOfExactType<MapStateInheritedWidget>()
+      ?.mapController;
+
+  static MapController of(BuildContext context) =>
+      maybeOf(context) ??
+      (throw StateError(
+          '`MapController.of()` should not be called outside a `FlutterMap` and its children'));
```

Comment:
Just a question. Does this mean that the controller can now be access by the end user in descendants, without having to specify a controller manually?

Reply:
This was previously possible with FlutterMapState.of(context).mapController. Now it is given a dedicated method to allow only looking up the MapController if that's all you need. This will also increase efficiency when the inherited widget is an inherited model since updating the state won't trigger an update of code which just wants the controller.